app/controllers/social_previews_controller.rb

```diff
@@ -62,4 +75,24 @@ def tag
       end
     end
   end
+
+  private
+
+  def define_categories
```

Comment:
Probably could have been pulled from the `ClassifiedListing` model as to not repeat here.